Formula/taskwarrior-tui.rb

```diff
@@ -0,0 +1,23 @@
+class TaskwarriorTui < Formula
+  desc "Terminal user interface for taskwarrior"
+  homepage "https://github.com/kdheepak/taskwarrior-tui"
+  url "https://github.com/kdheepak/taskwarrior-tui/archive/v0.8.0.tar.gz"
+  sha256 "15a2195dde970f369b5b1e346a087c72581142525765baeffe8d54b95824bae0"
+  license "MIT"
+  head "https://github.com/kdheepak/taskwarrior-tui"
+
+  livecheck do
+    url "https://github.com/kdheepak/taskwarrior-tui/releases/latest"
+    regex(%r{href=.*?/tag/v?(\d+(?:\.\d+)+)["' >]}i)
+  end
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system "#{bin}/taskwarrior-tui", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for the comment. This is a terminal user interface, so it doesn't have a flag other than `--help` and `--version` that run in command line only mode. Any suggestions for what I should do in this case would be appreciated.

Reply:
I've added a check for a command line option that fails when no argument is passed. I've also updated the check for the version number to check whether the string in the formula matches that of the output.

Reply:
Resolved in [`5fea297`](https://github.com/Homebrew/homebrew-core/pull/63373/commits/5fea297038cdaa5f063087c7c46e0dcc1e948b90).